### PR TITLE
Setup ktor unit testing with firebase emulators

### PR DESCRIPTION
**Pull request by matt989253**
This PR does the following:

- refactors and shares the firebase app initialisation from environment variables code
- adds a new workflow to test the web app on pull requests:
  - adds JUnit5 and Ktor testing dependencies
  - adds the Firestore Emulator to test in a controlled environment
  - adds two simple JUnit tests:
    - Verifying we are using the firebase emulator and not the production store
    - Verifying authentication as a PoC
    
PS: Alexandre Piveteau helped and Co-authored commits following ba22991a12484c658933660db60042bb466996a2.

### CODE REVIEW

File: .github/scripts/emulator-env.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+echo "#!/bin/bash"
+echo "export PATH=${JAVA_HOME}:\$PATH"
+echo "export JAVA_HOME=${JAVA_HOME}"
+echo "export GOOGLE_SERVICE_ACCOUNT=${GOOGLE_SERVICE_ACCOUNT@Q}"
+echo "export GOOGLE_DATABASE_NAME=${GOOGLE_DATABASE_NAME}"
+echo "export GOOGLE_BUCKET_NAME=${GOOGLE_BUCKET_NAME}"
+echo "export FIREBASE_TOKEN=${FIREBASE_TOKEN}"
+echo "export FIRESTORE_EMULATOR_HOST=localhost:8080"
+echo "./gradlew :web:test"
```
Comment (alexandrepiveteau) [inline .github/scripts/emulator-env.sh]:
```suggestion
echo "./gradlew :web:test"

```

**Comment by alexandrepiveteau** [comment]
image

**Review by alexandrepiveteau — DISMISSED** [review]

File: .github/workflows/web-pr.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        env:
+          GOOGLE_SERVICE_ACCOUNT: ${{ secrets.FIREBASE_GOOGLE_SERVICE_ACCOUNT }}
+          GOOGLE_DATABASE_NAME: ${{ secrets.FIREBASE_GOOGLE_DATABASE_NAME }}
+          GOOGLE_BUCKET_NAME: ${{ secrets.FIREBASE_GOOGLE_BUCKET_NAME }}
+          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+        run: |
+          chmod +x ../.github/scripts/emulator-env.sh
+          ../.github/scripts/emulator-env.sh > ../.github/scripts/run.sh
+          chmod +x ../.github/scripts/run.sh
+          sudo firebase emulators:exec --only firestore --project tupperdate-developement 'bash ../.github/scripts/run.sh'
```
Comment (glsubri) [inline .github/workflows/web-pr.yml]: I'm not sure what you are trying to do, but if you want to include a script from another, you should use the command `source` or in bash `.`. So in `run.sh` add a line:
```bash
. ../.github/scripts/emulator-env.sh
```

And then you should have access to all functions and variables, etc. In addition your `emulator-env.sh` does not need to be `chmod +x`.

If this is not what you're trying to do, I do not understand why you have two different scripts...?

**Review by glsubri — COMMENTED** [review]
The rest looks OK to me, but you know the API better than I do, so...

**Review by glsubri — APPROVED** [review]

**Pull request merged**